File: ecommerce-frontend/src/modules/order/services/paymentService.js
import api from '@/shared/utils/api'

export const paymentService = {
  /**
   * Get VNPay redirect URL for a given order.
   * @param {number} orderId - The order ID from the backend
   * @returns {Promise<{paymentUrl: string}>}
   */
  createVnpayPayment: async (orderId) => {
    return api.get(`/api/v1/payments/vnpay/create-payment?orderId=${orderId}`)
  },

  /**
   * Verify VNPay payment result by forwarding the return query params to backend.
   * @param {string} queryString - The full query string from VNPay redirect
   * @returns {Promise<{success: boolean, message: string, orderNumber?: string}>}
   */
  verifyVnpayPayment: async (queryString) => {
    return api.get(`/api/v1/payments/vnpay/payment-return?${queryString}`)
  },
}

File: backend/services/order-service/src/main/java/com/fashion/orderservice/service/PaymentService.java
package com.fashion.orderservice.service;

import java.util.Map;

public interface PaymentService {

    /**
     * Create a VNPay payment URL for the given order.
     *
     * @param orderId   the order's database ID
     * @param ipAddress the client's IP address
     * @return the VNPay redirect URL
     * @throws IllegalArgumentException if order not found or payment method is not
     *                                  VNPAY
     */
    String createPaymentUrl(Long orderId, String ipAddress);

    /**
     * Process VNPay payment callback: validate signature, publish payment result
     * event.
     *
     * @param params all query parameters from VNPay redirect
     * @return result map containing success status, orderId, orderNumber, etc.
     * @throws IllegalArgumentException if signature invalid, order not found, or
     *                                  txnRef invalid
     */
    Map<String, Object> processPaymentReturn(Map<String, String> params);
}
